### PR TITLE
3.1 updates

### PR DESCRIPTION
**Pull request by dzucconi**
### Colors
* `copper100` and `copper10` are removed

### Buttons
* `noOutline` becomes `tertiary`
* `secondaryOutline` becomes `secondaryBlack`
* `secondaryGray` becomes `primaryGray`
* `IconButton` becomes `Button` with the `Icon` prop

### Badges
* `Badge` becomes `Label`
* `defaultLight` becomes `light`
* `defaultDark` becomes `dark`

### Pills
* `textRound` becomes either `default` or `search` depending on application
* `textSquare` becomes `filter`

--------
## Colors

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @artsy/palette-charts@20.0.0-canary.1169.24204.0
  npm install @artsy/palette-tokens@4.0.0-canary.1169.24204.0
  npm install @artsy/palette@21.0.0-canary.1169.24204.0
  # or 
  yarn add @artsy/palette-charts@20.0.0-canary.1169.24204.0
  yarn add @artsy/palette-tokens@4.0.0-canary.1169.24204.0
  yarn add @artsy/palette@21.0.0-canary.1169.24204.0
  ```
</details>

### CODE REVIEW

**Comment by damassi** [comment]
From an ergonomics perspective, can we push back on those revised button labels? Is that necessary? 
- `noOutline` = "no outline". What does "tertiary" even mean? 
- `secondaryOutline` = `outline that is black`, not "another black button" which is the default

File: packages/palette/src/elements/Button/v3/tokens.ts [hunk 1]
```diff
@@ -1,158 +1,235 @@
...
       borderColor: "transparent",
-      color: "black30",
+      color: "blue100",
     },
   },
 }
 
+type ButtonVariants = Record<ButtonState, Record<ButtonVariant, unknown>>
+
+/** Inverts the structure of the style object so that we can use responsive values for `variant` */
```
Comment (damassi) [inline packages/palette/src/elements/Button/v3/tokens.ts:217]: I ran across this issue last week. Awesome 👍

**Comment by dzucconi** [comment]
### TODO

- [ ] Update `Color` type (possibly requires getting rid of v2 theme?)

**Comment by artsyit** [comment]
:rocket: PR was released in [`@artsy/palette-charts@20.0.0`](https://github.com/artsy/palette/releases/tag/%40artsy/palette-charts%4020.0.0), [`@artsy/palette-tokens@4.0.0`](https://github.com/artsy/palette/releases/tag/%40artsy/palette-tokens%404.0.0), [`@artsy/palette@21.0.0`](https://github.com/artsy/palette/releases/tag/%40artsy/palette%4021.0.0) :rocket:

**Pull request merged**